src/providers/hana/qgshanaproviderconnection.cpp

```diff
@@ -52,6 +54,12 @@ bool QgsHanaProviderResultIterator::hasNextRowPrivate() const
   return mNextRow;
 }
 
+long long QgsHanaProviderResultIterator::rowCountPrivate() const
+{
+  // TODO: hana team, this is for you.
```

Comment:
@mrylov Can you take care of the implementation and check whether we have or should implement additional methods?

Reply:
Sure, I will take a closer look.